x-pack/plugins/viz_editor/public/frame/expression_renderer/expression_renderer.tsx

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+// @ts-ignore
+import { fromExpression } from '@kbn/interpreter/common';
+import React, { useEffect, useRef } from 'react';
+
+async function runAndRender(
+  expression: any,
+  domElement: any,
+  getInterpreter: any,
+  renderersRegistry: any
+) {
+  const ast = fromExpression(expression);
+  const { interpreter } = await getInterpreter();
+  const response = await interpreter.interpretAst(ast, {}, { getInitialContext: () => ({}) });
+  if (response.type === 'render') {
+    renderersRegistry.get(response.as).render(domElement, response.value, {
+      onDestroy: (fn: () => never) => {
+        /* this is just here for compatibility with legacy kibana renderers */
+      },
+    });
+  }
+}
+
+export function ExpressionRenderer(props: any) {
+  const mountpoint: React.MutableRefObject<null | HTMLDivElement> = useRef(null);
+
+  useEffect(() => {
```

Comment:
This runs every time ExpressionRenderer is called, even if nothing in the expression has changed. I think you should call useEffect with a second argument to ensure it only runs when the arguments have changed.

Reply:
Makes sense, I added them